lib/tasks/deployment/20201005203140_populate_contact_type_groups_and_contact_types.rake

```diff
@@ -0,0 +1,51 @@
+namespace :after_party do
+  desc 'Deployment task: populate_contact_type_groups_and_contact_types'
+  task populate_contact_type_groups_and_contact_types: :environment do
+    puts "Running deploy task 'populate_contact_type_groups_and_contact_types'"
+
+    # Create Groups and Contact types for each existing organization
+    CasaOrg.all.each do |org|
+      ContactTypeGroup.create(casa_org: org, name: "CASA").tap do |group|
+        ContactType.create(contact_type_group: group, name: "Youth")
+        ContactType.create(contact_type_group: group, name: "Supervisor")
+      end
+
+      ContactTypeGroup.create(casa_org: org, name: "Biological family").tap do |group|
+        ContactType.create(contact_type_group: group, name: "Bio Parent")
+        ContactType.create(contact_type_group: group, name: "Other Family")
+      end
+
+      ContactTypeGroup.create(casa_org: org, name: "Placement").tap do |group|
+        ContactType.create(contact_type_group: group, name: "Foster Parent")
+      end
+
+      ContactTypeGroup.create(casa_org: org, name: "Social Services").tap do |group|
+        ContactType.create(contact_type_group: group, name: "Social Worker")
+        ContactType.create(contact_type_group: group, name: "DSS Worker")
+      end
+
+      ContactTypeGroup.create(casa_org: org, name: "Legal").tap do |group|
+        ContactType.create(contact_type_group: group, name: "Court")
+        ContactType.create(contact_type_group: group, name: "Attorney")
+      end
+
+      ContactTypeGroup.create(casa_org: org, name: "Health").tap do |group|
+        ContactType.create(contact_type_group: group, name: "Medical Professional")
+        ContactType.create(contact_type_group: group, name: "Therapist")
+      end
+
+      ContactTypeGroup.create(casa_org: org, name: "Education").tap do |group|
+        ContactType.create(contact_type_group: group, name: "School")
+      end
+
+      ContactTypeGroup.create(casa_org: org, name: "Support Worker").tap do |group|
+        ContactType.create(contact_type_group: group, name: "Other Support Worker")
+      end
+    end
```

Comment:
@roxannecojocariu to make sure this is idempotent, please change all the `create` statements to `first_or_create!`. Let's add the bang (!) so we can catch any failures.

Reply:
@littleforest I've made the suggested changes, however, none of the records are being created in my local environment now that I'm using `first_or_create!`. It looks like it's because of the seed file which creates two contact_types records.  From [what I've read ](https://stackoverflow.com/a/42178753/11535926)about `first_or_create!`, it will return the first "foo" if any are present.  So, in this case (and I think this will also be the case in QA and prod), no new contact_types or contact_type_groups will be created if even one record exists.

For this reason, I'm wondering if `find_or_create!` might be a better option.  What do you think?

Reply:
Oh yes, sorry! Use `find_or_create_by!`